src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2808,6 +2801,31 @@ void InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, int
             } while (i < TOKEN_HASH_SIZE);
             Contract.Assert(true, "The hashtable is full.  This should not happen.");
         }
+
+        private bool CompareStringOptimized(string string1, int offset1, int length1, string string2, int offset2, int length2)
+        {
+            if (length1 != length2)
+            {
+                return false;
+            }
+
+            // Optimize for one character cases which are common due to date and time separators (/ and :)
+            if (length1 == 1)
+            {
+                if (IsAscii(string1[offset1]) && IsAscii(string2[offset2]))
+                {
+                    return this.Culture.TextInfo.ToLower(string1[offset1]) == this.Culture.TextInfo.ToLower(string2[offset2]);
+                }
+            }
+
+            return (this.Culture.CompareInfo.Compare(string1, offset1, length1, string2, offset2, length2, CompareOptions.IgnoreCase) == 0);
+        }
+
+        // Return true for all characters below or equal U+007f, which is ASCII.
+        private static bool IsAscii(char ch)
+        {
+            return (ch <= '\x007f');
```

Comment:
We already have this method in several places in mscorlib.  It'd be nice to avoid introducing another.  Maybe make one of those other ones internal instead of private, e.g. Char.IsAscii?